components/camel-mongodb/src/main/java/org/apache/camel/component/mongodb/conf/MongoConfiguration.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.component.mongodb.conf;
+
+import java.util.Map;
+
+public class MongoConfiguration {
+
+    private String host = "localhost:27017";
+    private String user;
+    private String password;
+    private String adminDB = "admin";
+
+    public MongoConfiguration() {
+        super();
+    }
+
+    public MongoConfiguration(Map<String, String> parameters) {
+        super();
+        this.host = parameters.get("host");
+        this.user = parameters.get("user");
+        this.password = parameters.get("password");
+        // Optional parameters
+        String optionalAdminDB = parameters.getOrDefault("adminDB", "");
+        if (!"".equals(optionalAdminDB)) {
```

Comment:
Since you are checking for empty string, maybe we can change it like this:
```suggestion
        if (!optionalAdminDB.isEmpty()) {
```

Reply:
Yeah, I guess it improves readability, thanks.